code-challenges.test.js

```diff
@@ -14,43 +14,176 @@
+        // const oddOnlyArray = (array) => {
+        //     let newArr = []
+        //     for(let i=0; i<array.length; i++){
+        //         if(array[i] % 2 !== 0){
+        //             newArr.push(array[i]).sort
+        //         }
+        // //         }
+        //         return newArr.sort((a, b) => a - b),newArr.filter(array => typeof array == 'number')
+        //     }
+        //Function: oddOnlyArray, Variable: fullArr1, expected output
+        expect(oddOnlyArray(fullArr1)).toEqual([-9, 7, 9, 199])
+        //Function: oddOnlyArray, Variable: fullArr2, expected output
+        expect(oddOnlyArray(fullArr2)).toEqual([-823, 7, 23])
+      })
+})
+
+
+//Fail received: 
+//oddOnlyArray › takes in an array and returns a new array of only odd numbers sorted from least to greatest
+ //   ReferenceError: oddOnlyArray is not defined
 
 const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
-
-// b) Create the function that makes the test pass.
-
-
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// // b) Create the function that makes the test pass.
+//Function expression: oddOnlyArray
+     const oddOnlyArray = (array) => {
+            let newArr = []
+// Iterate over the length of the array
+            for(let i=0; i<array.length; i++){
+//Determine odd values
+                if(array[i] % 2 !== 0){
+//Sort odd values from least to greatest
+                    newArr.push(array[i]).sort
+                }
+            }
+//Use filter() method to filter out integers to avoid returning strings and booleans
+                return newArr.sort((a, b) => a - b),newArr.filter(array => typeof array == 'number')
+        }
+// _______________________________________________________________________________________________________________________________________
+
+// // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+//Function name: accumulatingArray
+//Function takes in an array and returns an array of the accumulating sum. An empty array should return an empty array. 
+// Variables: numbersToAdd1, numbersToAdd2, numbersToAdd3
+// Use map() method to return an array with the same amount of numbers 
+// What I used to better understand the math behind the accumulating sum: 
+//numbersToAdd1
+//at [0], add itself = 2
+//at [1], add itself + [0] = 6
+//at [2], add itself + [1] + [0] = 51 
+//at [3], add itself + [2] + [1] + [0] = 60
+
+//numbersToAdd2
+//at [0], add itself = 0
+//at [1], add itself + [0] = 7
+//at [2], add itself + [1] + [0] = -1
+//at [3], add itself + [2] + [1] + [0] = 11
```

Comment:
Loving this breakdown!!  This can be such a great strategy in an interview setting to help show that you understand exactly what needs to get done and helps direct your pseudocode even further.